sqlx-core/src/sqlite/options/mod.rs

```diff
@@ -171,23 +195,31 @@ impl SqliteConnectOptions {
     /// The default synchronous settings is FULL. However, if durability is not a concern,
     /// then NORMAL is normally all one needs in WAL mode.
     pub fn synchronous(mut self, synchronous: SqliteSynchronous) -> Self {
-        self.synchronous = synchronous;
+        self.pragmas
+            .insert("synchronous".into(), synchronous.as_str().into());
         self
     }
 
     /// Sets the [auto_vacuum](https://www.sqlite.org/pragma.html#pragma_auto_vacuum) setting for the database connection.
     ///
     /// The default auto_vacuum setting is NONE.
     pub fn auto_vacuum(mut self, auto_vacuum: SqliteAutoVacuum) -> Self {
-        self.auto_vacuum = auto_vacuum;
+        self.pragmas
+            .insert("auto_vacuum".into(), auto_vacuum.as_str().into());
+        self
+    }
+
+    /// Sets custom initial pragmas for the database connection.
+    pub fn pragma(mut self, key: &str, value: &str) -> Self {
```

Comment:
If you make the `Cow<'static, str>` change above, I would make `key` and `value` both take `impl Into<Cow<'static, str>>` here.